server/leafnode_test.go

```diff
@@ -4072,3 +4099,942 @@ func TestLeafNodeInterestPropagationDaisychain(t *testing.T) {
+listen: 127.0.0.1:-1
+accounts :{
+    A:{ jetstream: enabled, users:[ {user:a1,password:a1}]},
+	B:{ jetstream: enabled, users:[ {user:b1,password:b1}]}
+}
+jetstream : { domain: "DHUB", store_dir: "%s", max_mem: 100Mb, max_file: 100Mb }
+server_name: HUB2
+cluster: {
+	name: HUB
+	listen: 127.0.0.1:50555
+	routes=[nats-route://127.0.0.1:50554]
+}
+leafnodes: {
+	listen:127.0.0.1:-1
+}
+`
+
+	tmplL1 := `
+listen: 127.0.0.1:-1
+accounts :{
+    A:{ jetstream: enabled,  users:[ {user:a1,password:a1}]},
+	B:{ jetstream: disabled, users:[ {user:b1,password:b1}]}
+}
+jetstream: { domain: "DLEAF", store_dir: "%s", max_mem: 100Mb, max_file: 100Mb }
+server_name: LEAF1
+cluster: {
+	name: LEAF
+	listen: 127.0.0.1:50556
+	routes=[nats-route://127.0.0.1:50557]
+}
+leafnodes: {
+    remotes:[{url:nats://a1:a1@127.0.0.1:%d, account: A},{url:nats://b1:b1@127.0.0.1:%d, account: B}]
+}
+default_js_domain: {B:"DHUB"}
+`
+
+	tmplL2 := `
+listen: 127.0.0.1:-1
+accounts :{
+    A:{ jetstream: enabled,  users:[ {user:a1,password:a1}]},
+	B:{ jetstream: disabled, users:[ {user:b1,password:b1}]}
+}
+jetstream: { domain: "DLEAF", store_dir: "%s", max_mem: 100Mb, max_file: 100Mb }
+server_name: LEAF2
+cluster: {
+	name: LEAF
+	listen: 127.0.0.1:50557
+	routes=[nats-route://127.0.0.1:50556]
+}
+leafnodes: {
+    remotes:[{url:nats://a1:a1@127.0.0.1:%d, account: A},{url:nats://b1:b1@127.0.0.1:%d, account: B}]
+}
+default_js_domain: {B:"DHUB"}
+`
+
+	sd1 := createDir(t, JetStreamStoreDir)
+	defer os.RemoveAll(sd1)
+	confHub1 := createConfFile(t, []byte(fmt.Sprintf(tmplHub1, sd1)))
+	defer removeFile(t, confHub1)
+	sHub1, _ := RunServerWithConfig(confHub1)
+	defer sHub1.Shutdown()
+
+	sd2 := createDir(t, JetStreamStoreDir)
+	defer os.RemoveAll(sd2)
+	confHub2 := createConfFile(t, []byte(fmt.Sprintf(tmplHub2, sd2)))
+	defer removeFile(t, confHub2)
+	sHub2, _ := RunServerWithConfig(confHub2)
+	defer sHub2.Shutdown()
+
+	checkClusterFormed(t, sHub1, sHub2)
+	c1 := cluster{t: t, servers: []*Server{sHub1, sHub2}}
+	c1.waitOnPeerCount(2)
+
+	sd3 := createDir(t, JetStreamStoreDir)
+	defer os.RemoveAll(sd3)
+	confLeaf1 := createConfFile(t, []byte(fmt.Sprintf(tmplL1, sd3, sHub1.getOpts().LeafNode.Port, sHub1.getOpts().LeafNode.Port)))
+	defer removeFile(t, confLeaf1)
+	sLeaf1, _ := RunServerWithConfig(confLeaf1)
+	defer sLeaf1.Shutdown()
+
+	sd4 := createDir(t, JetStreamStoreDir)
+	defer os.RemoveAll(sd4)
+	confLeaf2 := createConfFile(t, []byte(fmt.Sprintf(tmplL2, sd3, sHub1.getOpts().LeafNode.Port, sHub1.getOpts().LeafNode.Port)))
+	defer removeFile(t, confLeaf2)
+	sLeaf2, _ := RunServerWithConfig(confLeaf2)
+	defer sLeaf2.Shutdown()
+
+	checkClusterFormed(t, sLeaf1, sLeaf2)
+	c2 := cluster{t: t, servers: []*Server{sLeaf1, sLeaf2}}
+	c2.waitOnPeerCount(2)
+
+	checkLeafNodeConnectedCount(t, sHub1, 4)
+	checkLeafNodeConnectedCount(t, sLeaf1, 2)
+	checkLeafNodeConnectedCount(t, sLeaf2, 2)
+
+	ncB := natsConnect(t, fmt.Sprintf("nats://b1:b1@127.0.0.1:%d", sLeaf1.getOpts().Port))
```

Comment:
I think `defer ncB.Close()` is missing here.

Reply:
fixed